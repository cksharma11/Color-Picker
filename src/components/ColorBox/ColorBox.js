import React, { useState } from "react";
import Clipboard from "react-clipboard.js";

const ColorBox = ({ hexCode, colorName }) => {
  const [status, setStatus] = useState(colorName);
  const setCopySuccess = () => {
    setStatus("Copied");
    setTimeout(() => {
      setStatus(colorName);
    }, 1000);
  };

  return (
    <Clipboard
      data-clipboard-text={hexCode}
      style={{ background: "none", border: "none", fontFamily: "inherit" }}
    >
      <div
        id={hexCode}
        className="div-colorbox"
        style={{ background: `${hexCode}` }}
        onClick={setCopySuccess}
      >
        {status}
      </div>
    </Clipboard>
  );
};

export default ColorBox;
