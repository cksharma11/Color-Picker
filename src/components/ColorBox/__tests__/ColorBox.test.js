import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import ColorBox from "../ColorBox";

Enzyme.configure({ adapter: new Adapter() });

describe("ColorBox", () => {
  it("should render properly", () => {
    const wrapper = shallow(<ColorBox haxCode="#bbb" />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
