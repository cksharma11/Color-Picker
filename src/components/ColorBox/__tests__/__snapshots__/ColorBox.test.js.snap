// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorBox should render properly 1`] = `
"<ClipboardButton data-clipboard-text=\\"#bbb\\" style={{...}} onClick={[Function: onClick]}>
  <div id=\\"#bbb\\" className=\\"div-colorbox\\" style={{...}} onClick={[Function: setCopySuccess]}>
    abc
  </div>
</ClipboardButton>"
`;
